src/testdrive/src/action/kafka/ingest.rs

```diff
@@ -133,30 +137,43 @@ impl Action for IngestAction {
                 key_schema,
                 value_schema,
             } => {
-                let schema_id = if self.publish {
+                let value_schema_id = if self.publish {
                     let ccsr_subject = format!("{}-value", topic_name);
                     let schema_id = state.tokio_runtime.block_on(
                         state
                             .ccsr_client
                             .publish_schema(&ccsr_subject, &value_schema)
                             .map_err(|e| format!("schema registry error: {}", e)),
                     )?;
-                    if let Some(key_schema) = key_schema {
+                    schema_id
+                } else {
+                    1
+                };
+                let key_encode = if let Some(key_schema) = key_schema {
+                    let schema = interchange::avro::parse_schema(&key_schema)
+                        .map_err(|e| format!("parsing avro schema: {}", e))?;
+                    if self.publish {
                         let key_subject = format!("{}-key", topic_name);
-                        state.tokio_runtime.block_on(
+                        let schema_id = state.tokio_runtime.block_on(
                             state
                                 .ccsr_client
                                 .publish_schema(&key_subject, &key_schema)
                                 .map_err(|e| format!("schema registry error: {}", e)),
                         )?;
+                        Some((schema, schema_id))
+                    } else {
+                        Some((schema, 2))
                     }
-                    schema_id
                 } else {
-                    1
+                    None
                 };
-                let schema = interchange::avro::parse_schema(&value_schema)
+                let value_schema = interchange::avro::parse_schema(&value_schema)
                     .map_err(|e| format!("parsing avro schema: {}", e))?;
-                Encoder::Avro { schema, schema_id }
+                Encoder::Avro {
+                    key_encode: if self.compacted { key_encode } else { None },
```

Comment:
Can we instead unconditionally set key_encode if it is present? I believe there is only one other place where the key schema is set, and that place should just be adjusted to ingest both a key and a value.

Reply:
How should the registry.td tests be supported? Do we copy the key out of the debezium envelope/record to make into a key? As in change line 159 from this
```
$ kafka-ingest format=avro topic=data schema=${schema-v1} key-schema=${valid-key-schema}
  publish=true timestamp=5
{"before": null, "after": {"a": 1}}
```

to this:
```
$ kafka-ingest format=avro topic=data schema=${schema-v1} key-schema=${valid-key-schema}
  publish=true timestamp=5
1 {"before": null, "after": {"a": 1}}
```
?

cc: @JLDLaughlin

Reply:
Yes, precisely. We should have been doing that all along. It just wasn’t strictly necessary for the purposes of the registry test.

Reply:
what should the key be when the record is `{"before": null, "after":null"}`? I just put `{"a":0}`

Reply:
I think we can just delete the sentinel record now. Shouldn’t be necessary with Natacha’s timestamping stuff.